react-app/src/features/package/index.tsx

```diff
@@ -51,16 +51,32 @@ export const DetailsContent: FC<{ id: string }> = ({ id }) => {
   );
 };
 
-export const Details = () => {
-  const { id, authority } = useParams<{
-    id: string;
-    authority: Authority;
-  }>();
+type LoaderData = {
+  id: string;
+  authority: Authority;
+};
 
+export const packageDetailsLoader = async ({ params }: LoaderFunctionArgs): Promise<LoaderData | Response> => {
+  const { id, authority } = params;
   if (id === undefined || authority === undefined) {
-    return <Navigate to="/dashboard" />;
+    return redirect("/dashboard");
+  }
+
+  try {
+    const packageResult = await getItem(id);
+    if (!packageResult || packageResult?._source?.deleted === true) {
+      return redirect("/dashboard");
+    }
+  } catch (error) {
+    console.log("Error: ", error);
```

Comment:
Would it make sense to expand on this slightly? Like `("Error in fetching packageResult: ", error)` Not sure what the exact message would be but just thinking this would be hard to pick out in a log as-is.

Reply:
I'd also add that maybe it's worth checking whether there's a `message` property on that error argument. `error instanceof Error` or some other similar check